packages/@n8n/db/src/entities/types-db.ts

```diff
@@ -272,8 +272,12 @@ export const enum StatisticsNames {
 export type AuthProviderType = 'ldap' | 'email' | 'saml'; // | 'google';
 
 export type FolderWithWorkflowAndSubFolderCount = Folder & {
-	workflowCount: boolean;
-	subFolderCount: number;
+	workflowCount?: boolean;
```

Comment:
Can we avoid changing these to optional? They should still be present always if folder is of this type

Reply:
I'm suggesting we simplify this, instead of introducing the new type can we change this type to

```
export type FolderWithWorkflowAndSubFolderCount = Folder & {
	workflowCount: boolean;
	subFolderCount: number;
	path?: string[];
};
```

and then enrich the path only if the select option is used, otherwise just leaving it out? This allows us to keep the `FolderRepository` typed as

```
export class FolderRepository extends Repository<FolderWithWorkflowAndSubFolderCount> {
```

and removes the need for casts

Reply:
The reason I did not want the repository to take `FolderWithWorkflowAndSubFolderCount` it's because does not have those values in the entity. So if you in this repo do: this.create, it looks like you can pass the fields in `FolderWithWorkflowAndSubFolderCount ` and that is not correct. The same with update. Those fields are only included if the getMany method and the select parameter explicitly include them. Thoughts?

![CleanShot 2025-05-08 at 18 47 38@2x](https://github.com/user-attachments/assets/a77456a9-459d-4e08-8073-8eea71d65d31)